app/Http/Controllers/Administration/UpdateController.php

```diff
@@ -118,16 +119,14 @@ public function view(): View
 	 * The whole code around installation/upgrade/migration should
 	 * thoroughly be revised an refactored.
 	 */
-	public function migrate(Request $request): View
+	public function migrate(Request $request): View|Response
 	{
-		$canExecute = Authorization::isAdmin() || Authorization::loginAsAdminIfNotRegistered();
+		$isLoggedIn = Authorization::check();
+		$isLoggedIn = $isLoggedIn || Authorization::loginAsAdminIfNotRegistered();
+		$isLoggedIn = $isLoggedIn || Legacy::loginAsAdmin($request->input('username', ''), $request->input('password', ''), $request->ip());
+		$isLoggedIn = $isLoggedIn || Authorization::loginAs($request->input('username', ''), $request->input('password', ''), $request->ip());
```

Comment:
I see that the decision whether someone is logged in and whether this someone is authorized to execute the update became more complicated than before. I don't doubt that this ia correct, but I cannot grasp it without thinking about it thoroughly. Please, add a comment above these lines which explain _each_ of these cases and why they are needed, i.e. the use case and how a user can get into that situation. (Not only a comment which states that we check for authorization, that is obvious).

As you stated on Gitter, you found a bug in a tricky edge case. It is great that you fixed it, but IHMO it is of equal importance to document it. Sooner or later someone will maybe refactor this code (it already contains the word "legacy") and this someone should be aware of what needs to be considered. In particular, as this amount of boolean conditions is not straightforward.

Reply:
> I see that the decision whether someone is logged in and whether this someone is authorized to execute the update became more complicated than before. 

I would beg to differ. The previous code made use of multiple if statements and used boolean to decide whether we could execute the migration. This is no longer the case: we have 2 steps: login - check & migrate.

Reply:
```php
		$canExecute = Authorization::isAdmin() || Authorization::loginAsAdminIfNotRegistered();
		if (!$canExecute &&
			!Legacy::loginAsAdmin($request['username'] ?? '', $request['password'] ?? '', $request->ip())
		   ) {
			Authorization::loginAs($request['username'] ?? '', $request['password'] ?? '', $request->ip());
			$canExecute = Authorization::isAdmin();
		}

		if ($canExecute) {
...
```
This is way harder to read that current version. The bug here is :
if you login as admin with `Legacy::loginAsAdmin()` we get `true` and the body of the loop is not executed.
As a result `$canExecute` is not set and the migration is not applied.

Reply:
can his be resolved ?

Reply:
Yes, as it is commented now.